File: engine/transpile/src/test/resources/golden/SYK006/WS_エラー商品テーブル.py
"""SYK006 のデータ部レコード WS-エラー商品テーブル を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_エラー商品テーブル:
    """01 WS-エラー商品テーブル 総バイト長 160。"""

    LENGTH = 160

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(160)
        else:
            self._data = bytearray(data)
            if len(self._data) < 160:
                self._data.extend(b"\x00" * (160 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-エラー商品
    def get_WS_エラー商品(self):
        return decode_alphanumeric(self._data, 0, 8)
    def set_WS_エラー商品(self, value):
        encode_alphanumeric(self._data, 0, 8, value)

File: src/engine/transpile/src/test/resources/golden/SYK006/WS_エラー商品テーブル.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK006 のデータ部レコード WS-エラー商品テーブル を逐語対訳した自動生成コード。 */
public final class WS_エラー商品テーブル {

    /** 01 WS-エラー商品テーブル 総バイト長 160。 */
    public static final int LENGTH = 160;

    private final byte[] data;

    public WS_エラー商品テーブル() {
        this.data = new byte[160];
    }

    public WS_エラー商品テーブル(byte[] source) {
        this.data = Arrays.copyOf(source, 160);
    }

    public byte[] data() {
        return data;
    }

    // WS-エラー商品
    public String get_WS_エラー商品() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 8);
    }
    public void set_WS_エラー商品(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 8, value);
    }
}
